src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs

```diff
@@ -469,9 +469,46 @@ public virtual RedirectToActionResult RedirectToAction(string actionName, string
             string controllerName,
             object routeValues)
         {
-            return new RedirectToActionResult(actionName, controllerName, routeValues)
+            return RedirectToAction(actionName, controllerName, routeValues, fragment: null);
+        }
+
+        /// <summary>
+        /// Redirects to the specified action using the specified <paramref name="actionName"/>,
+        /// <paramref name="controllerName"/>, and <paramref name="fragment"/>.
+        /// </summary>
+        /// <param name="actionName">The name of the action.</param>
+        /// <param name="controllerName">The name of the controller.</param>
+        /// <param name="fragment">The fragment to add to the URL</param>
+        /// <returns>The created <see cref="RedirectToActionResult"/> for the response.</returns>
+        [NonAction]
+        public virtual RedirectToActionResult RedirectToAction(
+            string actionName,
+            string controllerName,
+            string fragment)
+        {
+            return RedirectToAction(actionName, controllerName, routeValues: null, fragment: fragment);
+        }
+
+        /// <summary>
+        /// Redirects to the specified action using the specified <paramref name="actionName"/>,
+        /// <paramref name="controllerName"/>, <paramref name="routeValues"/>,
+        /// and <paramref name="fragment"/>.
+        /// </summary>
+        /// <param name="actionName">The name of the action.</param>
+        /// <param name="controllerName">The name of the controller.</param>
+        /// <param name="routeValues">The parameters for a route.</param>
+        /// <param name="fragment">The fragment to add to the URL</param>
+        /// <returns>The created <see cref="RedirectToActionResult"/> for the response.</returns>
+        [NonAction]
+        public virtual RedirectToActionResult RedirectToAction(
+            string actionName,
+            string controllerName,
+            object routeValues,
+            string fragment)
+        {
+            return new RedirectToActionResult(actionName, controllerName, routeValues, fragment)
             {
-                UrlHelper = Url,
+                UrlHelper = Url
```

Comment:
Did not know it was a style decision :) Thanks for pointing out, will do so in the future.